python/ray/serve/_private/deployment_state.py

```diff
@@ -1498,56 +1468,74 @@ def deploy(
             # Redeploying should not reset the deployment's start time.
             if not self._target_state.deleting:
-                deployment_info.start_time_ms = existing_info.start_time_ms
+                deployment_info.start_time_ms = curr_deployment_info.start_time_ms
 
             if (
                 not self._target_state.deleting
-                and existing_info.deployment_config == deployment_info.deployment_config
-                and existing_info.replica_config.ray_actor_options
+                and curr_deployment_info.deployment_config
+                == deployment_info.deployment_config
+                and curr_deployment_info.replica_config.ray_actor_options
                 == deployment_info.replica_config.ray_actor_options
                 and deployment_info.version is not None
-                and existing_info.version == deployment_info.version
+                and curr_deployment_info.version == deployment_info.version
+                and curr_deployment_info.target_capacity == target_capacity
+                and curr_deployment_info.target_capacity_scale_direction
+                == target_capacity_scale_direction
             ):
                 return False
 
-        # If autoscaling config is not none, decide initial num replicas
-        autoscaling_config = deployment_info.deployment_config.autoscaling_config
-        if autoscaling_config is not None:
-            adjust_capacity = False
-            if autoscaling_config.initial_replicas is not None and (
-                target_capacity_scale_direction is None
-                or target_capacity_scale_direction == TargetCapacityScaleDirection.UP
-            ):
-                autoscaled_num_replicas = autoscaling_config.initial_replicas
-            else:
-                if existing_info is not None:
-                    autoscaled_num_replicas = self._target_state.num_replicas
-                else:
-                    autoscaled_num_replicas = autoscaling_config.min_replicas
-            autoscaled_num_replicas = self.get_capacity_adjusted_num_replicas(
-                autoscaled_num_replicas, target_capacity
+        if curr_deployment_info is not None:
+            logger.info(
+                f"not exiting early: {target_capacity} "
+                f"{target_capacity_scale_direction} "
+                f"{curr_deployment_info.target_capacity} "
+                f"{curr_deployment_info.target_capacity_scale_direction}"
             )
-            deployment_info.set_autoscaled_num_replicas(autoscaled_num_replicas)
         else:
-            # When not autoscaling, defer adjusting the capacity to the update
-            # loop.
-            adjust_capacity = True
+            logger.info(
+                f"not exiting early: {target_capacity} "
+                f"{target_capacity_scale_direction} "
+            )
+
+        deployment_info.set_target_capacity(
+            target_capacity, target_capacity_scale_direction
+        )
```

Comment:
We need to pass in the `target_capacity` and `target_capacity_scale_direction` into the `deploy()` method anyways, so we can compare against the deployment's existing `target_capacity` and `scale_direction` and decide whether to exit early or not ([code source](https://github.com/shrekris-anyscale/ray/blob/7f1b8e91d2e3c1a831648b86812978cc7bec565a/python/ray/serve/_private/deployment_state.py#L1495-L1497)). I'd prefer to also set the target capacity in the deployment state to keep everything in one place.

Reply:
Could you instead do the following comparison to avoid having to pass them into the deploy method?
```
                and curr_deployment_info.target_capacity == deployment_info.target_capacity
                and curr_deployment_info.target_capacity_scale_direction
                == deployment_info.target_capacity_scale_direction
```

Reply:
Nice, yeah that's a great idea. I set the `target_capacity` and `target_capacity_scale_direction` in the application state manager.